packages/notify-client/test/clients.spec.ts

```diff
@@ -841,5 +848,63 @@ describe("Notify", () => {
         expect(Object.keys(wallet3.getActiveSubscriptions()).length).toEqual(0);
       });
     });
+
+    describe.skipIf(!hasTestProjectSecret)("Message Deduping", () => {
+      it("dedups messages based on notify message id", async () => {
+        await createNotifySubscription(wallet, account, onSign);
+
+        expect(wallet.subscriptions.getAll().length).toEqual(1);
+
+        const testSub = wallet.subscriptions.getAll()[0];
+
+        expect(
+          Object.keys(wallet.messages.get(testSub.topic).messages).length
+        ).toEqual(0);
+
+        let messagesReceived = 0;
+
+        wallet.on("notify_message", () => {
+          messagesReceived++;
+        });
+
+        const now = Date.now();
+
+        await waitForEvent(() => Date.now() - now > 1_000);
+
+        let receivedRealMessage = false;
+        let message: any = {};
+        wallet.on("notify_message", (m) => {
+          receivedRealMessage = true;
+          message = m;
+        });
+
+        await sendNotifyMessage(account, "Test");
+
+        await waitForEvent(() => receivedRealMessage);
+
+        const encoded = await wallet.core.crypto.encode(testSub.topic, {
+          ...message,
+          // Set different JSONRPC ID
```

Comment:
Was trying to work out why this is being done at first, so adding the (presumed) reasoning:

```suggestion
          // Set different JSONRPC ID to avoid the relayer deduping the message based on JSON-RPC ID
```

Reply:
This is explained in the comments below, wouldn't adding this be redundant?

Reply:
Action (override ID) and intent (avoid eager deduping) is clear but not mechanism (relayer will interfere here if not handled).

Not super important but felt this may be unclear to someone reading this if they’re not intimately familiar with the relayer logic)

Reply:
Ah I didn't consider that, Yep I'll commit your suggestion